### PR TITLE
Delete multiple definition of functions for gcc 10.2.1

### PR DESCRIPTION
**Pull request by Jamaika1**

### CODE REVIEW

File: src_base/xeve_itdq.h [hunk 1]
```diff
@@ -47,7 +47,7 @@
 void xeve_sub_block_itdq(s16 coef[N_C][MAX_CU_DIM], int log2_cuw, int log2_cuh, u8 qp_y, u8 qp_u, u8 qp_v, int flag[N_C], int nnz_sub[N_C][MAX_SUB_TB_NUM], int bit_depth);
 
 typedef void (*INV_TRANS)(s16 *, s16 *, int, int, int, int);
-INV_TRANS (*xeve_func_itrans)[16][5];
+static INV_TRANS (*xeve_func_itrans)[16][5];
```
Comment (MoonkiHong) [inline src_base/xeve_itdq.h:50]: @Jamaika1 Could you provide the details of your PR specifically restate several functions with its new `static` type? My humble opinion is to revise the title of this PR more properly with several subordinate descriptions for the better readability of this PR.

File: src_base/xeve_mc.h [hunk 2]
```diff
@@ -90,9 +90,9 @@ typedef void (*XEVE_AVG_NO_CLIP)(s16 *src, s16 *ref, s16 *dst, int s_src, int s_
 extern XEVE_MC_L xeve_tbl_mc_l[2][2];
 extern XEVE_MC_C xeve_tbl_mc_c[2][2];
 
-XEVE_MC_L (*xeve_func_mc_l)[2][2];
-XEVE_MC_C (*xeve_func_mc_c)[2][2];
-XEVE_AVG_NO_CLIP xeve_func_average_no_clip;
+static XEVE_MC_L (*xeve_func_mc_l)[2][2];
```
Comment (MoonkiHong) [inline src_base/xeve_mc.h:93]: @Jamaika1 Please refer to my comments above.

File: src_base/xeve_sad.h [hunk 3]
```diff
@@ -48,10 +48,10 @@ extern XEVE_FN_SSD xeve_tbl_ssd_16b[8][8];
 extern XEVE_FN_DIFF xeve_tbl_diff_16b[8][8];
 extern XEVE_FN_SATD xeve_tbl_satd_16b[1];
 
-XEVE_FN_SAD (* xeve_func_sad)[8][8];
-XEVE_FN_SSD (* xeve_func_ssd)[8][8];
-XEVE_FN_DIFF (* xeve_func_diff)[8][8];
-XEVE_FN_SATD (* xeve_func_satd)[1];
+static XEVE_FN_SAD (* xeve_func_sad)[8][8];
```
Comment (MoonkiHong) [inline src_base/xeve_sad.h:51]: @Jamaika1 Please refer to the comments above.

File: src_base/xeve_tq.h [hunk 4]
```diff
@@ -39,7 +39,7 @@
 typedef void(*XEVE_TXB)(void * coef, void * t, int shift, int line, int step);
 extern XEVE_TXB xeve_tbl_txb[MAX_TR_LOG2];
 extern const int quant_scale[6];
-s64 err_scale_tbl[6][NUM_CU_LOG2 + 1];
+static s64 err_scale_tbl[6][NUM_CU_LOG2 + 1];
```
Comment (MoonkiHong) [inline src_base/xeve_tq.h:42]: @Jamaika1 Please refer to the comments above.

File: src_main/xevem_ipred.h [hunk 5]
```diff
@@ -63,6 +63,6 @@ void xevem_get_mpm(int x_scu, int y_scu, int cuw, int cuh, u32 * map_scu, s8 * m
 
 typedef void(*XEVE_INTRA_PRED_ANG)(pel *src_le, pel *src_up, pel *src_ri, u16 avail_lr, pel *dst, int w, int h, int ipm, int bit_depth);
 extern XEVE_INTRA_PRED_ANG xeve_tbl_intra_pred_ang[3][2];
-XEVE_INTRA_PRED_ANG(*xeve_func_intra_pred_ang)[3][2];
+static XEVE_INTRA_PRED_ANG(*xeve_func_intra_pred_ang)[3][2];
```
Comment (MoonkiHong) [inline src_main/xevem_ipred.h:66]: @Jamaika1 Please refer to the comments above.

File: src_main/xevem_itdq.h [hunk 6]
```diff
@@ -44,7 +44,7 @@ void xeve_init_multi_tbl();
 void xeve_init_multi_inv_tbl();
 
 typedef void (*INV_TRANS)(s16 *, s16 *, int, int, int, int);
-INV_TRANS (*xeve_func_itrans)[16][5];
+static INV_TRANS (*xeve_func_itrans)[16][5];
```
Comment (MoonkiHong) [inline src_main/xevem_itdq.h:47]: @Jamaika1 Please refer to the comments above.

File: src_main/xevem_mc.h [hunk 7]
```diff
@@ -52,9 +52,9 @@ extern XEVE_DMVR_MC_L xevem_tbl_dmvr_mc_l[2][2];
 extern XEVE_DMVR_MC_C xevem_tbl_dmvr_mc_c[2][2];
 extern XEVE_MC_C      xevem_tbl_bl_mc_l[2][2];
 
-XEVE_DMVR_MC_L (*xevem_func_dmvr_mc_l)[2][2];
-XEVE_DMVR_MC_C (*xevem_func_dmvr_mc_c)[2][2];
-XEVE_MC_C      (*xevem_func_bl_mc_l)[2][2];
+static XEVE_DMVR_MC_L (*xevem_func_dmvr_mc_l)[2][2];
```
Comment (MoonkiHong) [inline src_main/xevem_mc.h:55]: @Jamaika1 Please refer to the comments above.

File: src_main/xevem_mc.h [hunk 8]
```diff
@@ -90,8 +90,8 @@ typedef void (*XEVE_AFFINE_H_SOBEL_FLT)(pel *pred, int pred_stride, int *derivat
 typedef void (*XEVE_AFFINE_V_SOBEL_FLT)(pel *pred, int pred_stride, int *derivate, int derivate_buf_stride, int width, int height);
 typedef void (*XEVE_AFFINE_EQUAL_COEF)(pel *residue, int residue_stride, int **derivate, int derivate_buf_stride, s64(*equal_coeff)[7], int width, int height, int vertex_num);
 
-XEVE_AFFINE_H_SOBEL_FLT xevem_func_aff_h_sobel_flt;
-XEVE_AFFINE_V_SOBEL_FLT xevem_func_aff_v_sobel_flt;
-XEVE_AFFINE_EQUAL_COEF  xevem_func_aff_eq_coef_comp;
+static XEVE_AFFINE_H_SOBEL_FLT xevem_func_aff_h_sobel_flt;
```
Comment (MoonkiHong) [inline src_main/xevem_mc.h:93]: @Jamaika1 Please refer to the comments above.

File: src_main/xevem_type.h [hunk 9]
```diff
@@ -36,7 +36,7 @@
 #ifndef _XEVEM_TYPE_H_
 #define _XEVEM_TYPE_H_
 
-#include <xeve_type.h>
+#include "xeve_type.h"
```
Comment (MoonkiHong) [inline src_main/xevem_type.h:39]: @Jamaika1 Seems this is not related to the main goal of your PR at this moment. For a better `git log` historical management for our community, what about separating this commit as a separate PR?

File: src_base/xeve_port.h [hunk 10]
```diff
@@ -141,9 +141,9 @@ void xeve_trace_line(char * pre);
 #define xeve_assert_gv(x,r,v,g) \
     {if(!(x)){assert(x); (r)=(v); goto g;}}
 
-
-
+#ifdef __SSE4.1__
```
Comment (MoonkiHong) [inline src_base/xeve_port.h:144]: @Jamaika1 This seems not related to the main goal of this PR (in touch with the PR title). What about separating this commit as a new PR?

**Review by MoonkiHong — COMMENTED** [review]
@Jamaika1 Thank you for your kind and passionate contributions. Please take a look at my humble opinions about your PR.

File: src_base/xeve_itdq.h (hunk 1 above)
Comment (Jamaika1) [reply src_base/xeve_itdq.h:50]: Hello. I wanted to test the new evc codec in GCC 10.2.1 under Windsows.
I used the function standard under gcc -std = gnu11 -ftree-vectorize -g0 -O3 -fPIC and couldn't compile the codec.
```
xevem_df.o:xevem_df.c:(.bss+0x0): multiple definition of `xevem_func_bl_mc_l'; xevem.o:xevem.c:(.bss+0xc8): first defined here
```

Comment (Jamaika1) [reply src_base/xeve_itdq.h:50]: @MoonkiHong Not all defined functions could be compiled in config. Some only worked.
```
-DWINDOWS_MUTEX_SYNC -DENC_ECU_ADAPTIVE -DMULTI_REF_ME_STEP -DENABLE_IBC_CHROMA_REFINE -DNET_AMVP -DET_ME_REFIDX1 -DATS_INTRA_FAST -DHTDF_FAST_TBL
```

File: src_base/xeve_port.h (hunk 10 above)
Comment (Jamaika1) [reply src_base/xeve_port.h:144]: To compile the codec I had to disable all functions above SSE2. Does the codec work without SIMD?

File: src_base/xeve_itdq.h (hunk 1 above)
Comment (Jamaika1) [reply src_base/xeve_itdq.h:50]: @MoonkiHong I compiled the codec for Windows. Unfortunately when producing movies the codec only creates the evc container header.

Comment (MoonkiHong) [reply src_base/xeve_itdq.h:50]: @Jamaika1 Sounds reasonable. We will check out your PR shortly.

File: src_base/xeve_port.h (hunk 10 above)
Comment (MoonkiHong) [reply src_base/xeve_port.h:144]: @Jamaika1 Thank you for rectifying your commit. We will check out your PR and get back to you soon.

**Comment by mpeg5** [comment]
Hi, all,
In order to support gcc 10.02, I think it would be required to change the variable declaration parts rather than just adding 'static' on each variables. Since this project should not have any issue regarding compiler version, we will fix this issue and commit new code soon. 
Thanks all.

**Comment by MoonkiHong** [comment]
> Hi, all,
> In order to support gcc 10.02, I think it would be required to change the variable declaration parts rather than just adding 'static' on each variables. Since this project should not have any issue regarding compiler version, we will fix this issue and commit new code soon.
> Thanks all.

@Jamaika1 What about creating a [dedicate issue](https://github.com/mpeg5/xeve/issues), so that we all can follow it up once the new commit comes?

**Review by MoonkiHong — COMMENTED** [review]
@Jamaika1 Please check out our latest feedback and suggestion. Thank you for your passionate contribution.

**Comment by mpeg5** [comment]
@Jamaika1 
Could you check the compilation issue again after pulling latest code?
